include/convex_bodies/hpolytope.h

```diff
@@ -1010,9 +1010,10 @@ class HPolytope {
 
     // updates the velocity vector v and the position vector p after a reflection
     // the real value of p is given by p + moved_dist * v
+    // MT must be sparse, in RowMajor format
     template <typename update_parameters>
     auto compute_reflection(Point &v, Point &p, update_parameters const& params) const
-         -> std::enable_if_t<std::is_same_v<MT, Eigen::SparseMatrix<NT, Eigen::RowMajor>> && !std::is_same_v<update_parameters, int>, void> { // MT must be in RowMajor format
+         -> std::enable_if_t<std::is_same_v<MT, Eigen::SparseMatrix<NT, Eigen::RowMajor>> && !std::is_same_v<update_parameters, int>, void> {
```

Comment:
What is `update_parameters` here? I do not understand `!std::is_same_v<update_parameters` could you please explain?

Reply:
basically there's another compute_reflection function above which takes an integer (just the facet) as the 3rd argument, and for some reason, if I don't have that condition the compiler decides to call this function assuming that the typename of update_parameters is int. There might maybe be better ways of dealing with these issues, but I remember I tried to solve them for some time and this is the best I could do, I couldn't at all understand how the compiler chooses which function to use when there's multiple ones that match

Reply:
It seems that we are too generic here and that complicates the interface a lot! `compute_reflection` is called by several walks. Instead of creating all those complicated overloads why not simply create a new name. Especially if this is the case that this function is only called by a single walk (i.e. accelerated billiard walk). Moreover, `update_parameters` should not be a template but the struct defined in billiard walk, in all other cases this code will not compile since all other "update_parameters" does not have a `moved_dist` field.